modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/impl/StandaloneMetaStorageManager.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.ignite.internal.vault.VaultManager;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.ClusterService;
+import org.apache.ignite.network.NetworkAddress;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+
+/**
+ * MetaStorageManager dummy implementation.
+ */
+@TestOnly
+public class StandaloneMetaStorageManager extends MetaStorageManagerImpl {
+    private static final String TEST_NODE_NAME = "test";
+
+    /**
+     * Creates standalone MetaStorage manager for provided VaultManager.
+     */
+    public static StandaloneMetaStorageManager create(VaultManager vaultMgr) {
+        return create(vaultMgr, new SimpleInMemoryKeyValueStorage(TEST_NODE_NAME));
+    }
+
+    /**
+     * Creates standalone MetaStorage manager for provided VaultManager.
+     */
+    public static StandaloneMetaStorageManager create(VaultManager vaultMgr, KeyValueStorage storage) {
+        return new StandaloneMetaStorageManager(
+                vaultMgr,
+                Mockito.mock(ClusterService.class),
+                Mockito.mock(ClusterManagementGroupManager.class),
+                Mockito.mock(LogicalTopologyService.class),
+                Mockito.mock(RaftManager.class),
+                storage
+        );
+    }
+
+    private final KeyValueStorage keyValueStorage;
+
+    /**
+     * The constructor.
+     *
+     * @param vaultMgr Vault manager.
+     * @param clusterService Cluster network service.
+     * @param cmgMgr Cluster management service Manager.
+     * @param logicalTopologyService Logical topology service.
+     * @param raftMgr Raft manager.
+     * @param storage Storage. This component owns this resource and will manage its lifecycle.
+     */
+    public StandaloneMetaStorageManager(VaultManager vaultMgr, ClusterService clusterService, ClusterManagementGroupManager cmgMgr,
+            LogicalTopologyService logicalTopologyService, RaftManager raftMgr, KeyValueStorage storage) {
+        super(vaultMgr, clusterService, cmgMgr, logicalTopologyService, raftMgr, storage);
+
+        keyValueStorage = storage;
+    }
+
+    @Override
+    public void start() {
+        keyValueStorage.start();
+
+        MetaStorageListener metaStorageListener = new MetaStorageListener(keyValueStorage);
+
+        RaftGroupService raftGroupService = Mockito.mock(RaftGroupService.class);
+        Mockito.when(raftGroupService.run(ArgumentMatchers.any()))
+                .thenAnswer(invocation -> runCommand(invocation.getArgument(0), metaStorageListener));
+
+        var localNode = new ClusterNode(TEST_NODE_NAME, TEST_NODE_NAME, new NetworkAddress("localhost", 10000));
+
+        MetaStorageServiceImpl service = new MetaStorageServiceImpl(raftGroupService, new IgniteSpinBusyLock(), localNode);
+
+        metaStorageServiceFuture().complete(service);
```

Comment:
This looks like a hack, I would like to propose a more clean solution using mocks, this way we won't have to complete the future explicitly, but it will be completed with our mock RaftGroupService:
```
private static RaftManager mockRaftManager() {
    ArgumentCaptor<RaftGroupListener> listenerCaptor = ArgumentCaptor.forClass(RaftGroupListener.class);

    RaftManager raftManager = mock(RaftManager.class);

    RaftGroupService raftGroupService = mock(RaftGroupService.class);

    try {
        when(raftManager.startRaftGroupNode(any(), any(), listenerCaptor.capture(), any()))
                .thenReturn(completedFuture(raftGroupService));
    } catch (NodeStoppingException e) {
        throw new RuntimeException(e);
    }

    when(raftGroupService.run(any()))
            .thenAnswer(invocation -> {
                Command command = invocation.getArgument(0);

                RaftGroupListener listener = listenerCaptor.getValue();

                return runCommand(command, listener);
            });

    return raftManager;
}
```

Reply:
I just wanted to avoid all RAFT related stuff.
Your approach is a bit complex and allow to emulate MetaStorage nodes and learner, I guess.

If you see it useful, ok, I'll fix this.

Reply:
yes, it's still more straightforward than completing an internal future explicitly

Reply:
Ok. Anyway, this will significantly reduce code complexity in tests where this manager will be used.